internal/events/pubsub.go

```diff
@@ -3,48 +3,29 @@ package events
 import (
 	gcppubsub "cloud.google.com/go/pubsub"
 	"context"
-	"github.com/googleapis/google-cloudevents-go/cloud/auditdata"
 	"go.uber.org/zap"
-	"google.golang.org/protobuf/encoding/protojson"
 )
 
-type notifier struct {
-	t   Subscription
+type subscription struct {
+	t   *gcppubsub.Subscription
 	log *zap.SugaredLogger
 }
 
 type Subscription interface {
-	Receive(ctx context.Context, f func(context.Context, *gcppubsub.Message)) error
+	Receive(ctx context.Context, event chan<- *gcppubsub.Message)
 }
 
-func (n *notifier) Receive(ctx context.Context, event chan<- InterruptionEvent) {
-	err := n.t.Receive(ctx, func(ctx context.Context, m *gcppubsub.Message) {
+func (s *subscription) Receive(ctx context.Context, event chan<- *gcppubsub.Message) {
+	err := s.t.Receive(ctx, func(ctx context.Context, m *gcppubsub.Message) {
```

Comment:
It's probably worth a comment here to explain what the behaviour is - presumably `s.t.Receive` blocks and continues calling the handler function for every message until the context is cancelled?

Reply:
Correct - added as a comment on the method itself